google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/AppEngineDeploy.java

```diff
@@ -80,42 +80,40 @@ public void deploy(
     final StringBuilder rawDeployOutput = new StringBuilder();
 
     DefaultDeployConfiguration configuration = new DefaultDeployConfiguration();
+    String yamlName =
+        deploymentConfiguration
+            .getEnvironment()
+            .equals(AppEngineEnvironment.APP_ENGINE_STANDARD.name())
+            || deploymentConfiguration
+            .getEnvironment()
+            .equals(AppEngineEnvironment.APP_ENGINE_FLEX_COMPAT.name())
+            ? "app.yaml"
+            : Paths.get(deploymentConfiguration.getYamlPath()).getFileName().toString();
```

Comment:
Not if the app.yaml files aren't called app.yaml, but e.g., custom.yamln java.yaml, etc.